presto-parser/src/test/java/com/facebook/presto/sql/TestSqlTemplateFormatter.java

```diff
@@ -0,0 +1,158 @@
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testRewriteGenericLiteral()
+    {
+        Statement statement = new Call(QualifiedName.of("foo"), ImmutableList.of(
+                new CallArgument("a", new GenericLiteral("varchar", "test"))));
+        String templateStatement = "CALL foo(a => ?)";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testSqlFormatterIntervalLiteral()
+    {
+        String statement = "SELECT * FROM table1 where Date1 < Date2 + INTERVAL '33' day to second";
+        String templateStatement = "SELECT * FROM table1 where Date1 < Date2 + ?";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testSqlFormatterNullLiteral()
+    {
+        assertTemplate("SELECT * FROM table1 WHERE null", "SELECT * FROM table1 WHERE ?");
+    }
+
+    @Test
+    public void testSqlFormatterStringLiteral()
+    {
+        Statement statement = new Call(QualifiedName.of("foo"), ImmutableList.of(
+                new CallArgument("b", new StringLiteral("go"))));
+        String templateStatement = "CALL foo(b => ?)";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testSqlFormatterTimeLiteral()
+    {
+        assertTemplate("SELECT * FROM table1 WHERE Time = '01:02:03.1234'", "SELECT * FROM table1 WHERE Time = ?");
+    }
+
+    @Test
+    public void testSqlFormatterTimestampLiteral()
+    {
+        assertTemplate("SELECT * FROM table1 WHERE DateTime > '2022-01-01 01:02:03.1234'", "SELECT * FROM table1 WHERE DateTime > ?");
+    }
+
+    @Test
+    public void testSqlFormatterMultipleLiteralTypesWithInnerJoinWithWhereWithGroupByWithHaving()
+    {
+        String statement = "SELECT table1.id, AVG(table2.price) FROM table1 INNER JOIN table2 ON table1.id = table2.id WHERE table2.state = 'CA' " +
+                "AND table2.registered = true AND null GROUP BY table1.id HAVING AVG(price) > 20 AND registrationDate <= '2022-01-01 01:02:03.1234'";
+        String templateStatement = "SELECT table1.id, AVG(table2.price) FROM table1 INNER JOIN table2 ON table1.id = table2.id WHERE table2.state = ? " +
+                "AND table2.registered = ? AND ? GROUP BY table1.id HAVING AVG(price) > ? AND registrationDate <= ?";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testSqlFormatterMultipleLiteralTypesWithWhereWithIntersectWithHavingWithOrderBy()
+    {
+        String statement = "SELECT id FROM table1 WHERE id > 78 INTERSECT SELECT id" +
+                "FROM table2 WHERE quantity <> 0 HAVING COUNT(*) > 1 ORDER BY table2.timestamp ASC";
+        String templateStatement = "SELECT id FROM table1 WHERE id > ? INTERSECT SELECT id" +
+                "FROM table2 WHERE quantity <> ? HAVING COUNT(*) > ? ORDER BY table2.timestamp ASC";
+        assertTemplate(statement, templateStatement);
+    }
+
```

Comment:
Can we add more tests for complex SQL to make sure we are getting good coverage on all the possible SQL statements that this formatter might encounter?